test/functional/http/post/5.dapps.js

```diff
@@ -0,0 +1,503 @@
+'use strict';
+
+var node = require('../../../node');
+var shared = require('../../shared');
+var constants = require('../../../../helpers/constants');
+
+var sendTransactionPromise = require('../../../common/apiHelpers').sendTransactionPromise;
+var creditAccountPromise = require('../../../common/apiHelpers').creditAccountPromise;
+var waitForConfirmations = require('../../../common/apiHelpers').waitForConfirmations;
+
+describe('POST /api/transactions (type 5) register dapp', function () {
+
+	var transaction;
+	var transactionsToWaitFor = [];
+	var badTransactions = [];
+	var goodTransactions = [];
+	var badTransactionsEnforcement = [];
+	var goodTransactionsEnforcement = [];
+
+	var account = node.randomAccount();
+	var accountNoFunds = node.randomAccount();
+	var accountMinimalFunds = node.randomAccount();
+
+	var dappDuplicate = node.randomApplication();
+	var dappDuplicateNameSuccess = node.randomApplication();
+	var dappDuplicateNameFail = node.randomApplication();
+	dappDuplicateNameSuccess.name = dappDuplicateNameFail.name;
```

Comment:
Move to the before section

Reply:
and kindly explain why we need to do this.